### ISSUE TITLE
After finishing a very long running statement the transaction is rolled back after 10 minutes while this features is unchecked

### ISSUE THREAD

**Issue by nilsvandermeer** [dbeaver/dbeaver#18497]
### Description

Now for third time, so it is a bug. Multiple version (happened first time 4 weeks ago).

Case:
Postgresql function runs for 1h45m. I come back, see a result in the Data Editor, Commit. No Error message or anything.
But: nothing has been done because the transaction has been rolled back.

From dbeaver_sql.log:
!ENTRY org.jkiss.dbeaver.model 1 53 2022-12-11 11:44:45.416
!MESSAGE select eyeint.createCoproductieCorporatie()
!SUBENTRY 1 org.jkiss.dbeaver.model 1 0 2022-12-11 13:29:51.435
!MESSAGE SUCCESS [-1]

In dbeaver-debug.log: 
2022-12-11 13:39:48.817 - End idle 1 transactions for postgres-jdbc-1729e7ddbc1-66aad6f3e00553c4

Automatically end long idle transactions is unchecked, for dbeaver as well as the connection.

### DBeaver Version

Community edition Version 22.3.0.202212041619

### Operating System

Windows 10

### Database and driver

PostgreSQL 9.4.22
Driver: 42.5.0

### Steps to reproduce

Connect to the database
Execute a very long running function (1h45m)
Wait for 10 minutes after the function has finished running
Idle transaction is ended, all the work is lost

### Additional context

Office extension installed

**Comment by LonwoLonwo** [comment]
Hello @nilsvandermeer 

Can you provide screens of these settings, please?

![2022-12-12 12_44_40-Window](https://user-images.githubusercontent.com/45152336/207013714-cb1107cf-3687-4c4e-b5ef-8a93ee08cf8a.png)

![2022-12-12 12_44_53-Window](https://user-images.githubusercontent.com/45152336/207013748-85940b73-057d-4ffa-ab93-00d4d6af7700.png)

![2022-12-12 12_45_17-Window](https://user-images.githubusercontent.com/45152336/207013768-7b0b8828-72f3-4140-bae5-a75399e3fb89.png)

**Comment by nilsvandermeer** [comment]
Hello Anastasiya,

Here are the settings for the connection:

***@***.***

***@***.***

***@***.***

**Comment by LonwoLonwo** [comment]
Hello @nilsvandermeer 

You can not attach files via email. Please provide images here, on the GitHub page.

**Comment by nilsvandermeer** [comment]
![image](https://user-images.githubusercontent.com/63210253/207294055-1269570b-ca29-4eac-a285-f705938498f6.png)
![image](https://user-images.githubusercontent.com/63210253/207294140-9d1aec34-e80f-4088-b1bd-d37029009c8f.png)
![image](https://user-images.githubusercontent.com/63210253/207294161-a847dee9-c69e-4cd5-bb1e-cd781a20784f.png)

**Comment by michal-bartak** [comment]
To me, it looks like the transaction has been rolled back because you left it open in the idle state for too long (which is not good practice)
It doesn't have to be DBeaver setting. It might be the server setting. Check out `idle_in_transaction_session_timeout` for your session.

**Comment by nilsvandermeer** [comment]
@michalk-k the settings you mention was introduced in postgresql 9.6, and I work with 9.4.22.
And in that case I would also not expect in dbeaver-debug.log:
2022-12-11 13:39:48.817 - End idle 1 transactions for postgres-jdbc-1729e7ddbc1-66aad6f3e00553c4

And in production environment I would not leave a transaction open that long, but in development I have no problems with it.

**Comment by michal-bartak** [comment]
Yeah, you are right.  Then it might be related to recently added auto disconnections of idle connections issued by DBeaver itself.

What I can advise more is to check if you have a separate connection for meta information or not. If SQL editor shares a connection with meta info, those may influence somehow.

![image](https://user-images.githubusercontent.com/7868445/208696551-2c1447c2-4925-4510-bdc5-871c527dab20.png)

**Comment by nilsvandermeer** [comment]
![image](https://user-images.githubusercontent.com/63210253/208700202-46ec6d90-53d0-4813-b86a-75b04293f143.png)

And when I look at sessions, I see a separate connection for metadata:
![image](https://user-images.githubusercontent.com/63210253/208700570-53d7e76f-cc66-4f06-b056-34287694be5e.png)
